table/metadata.go

```diff
@@ -134,6 +134,19 @@ type Metadata interface {
 	NameMapping() iceberg.NameMapping
 
 	LastSequenceNumber() int64
+	// StatisticsFile is an optional list of table statistics.
+	// Table statistics files are valid Puffin files.
+	// StatisticsFile are informational. A reader can choose to ignore statistics information.
+	// StatisticsFile support is not required to read the table correctly.
+	// A table can contain many statistics files associated with different table snapshots.
+	Statistics() []StatisticsFile
```

Comment:
should we return an `iter.Seq[StatisticsFile]` so that the slice isn't able to be modified by users?